lms/security.py

```diff
@@ -87,13 +87,25 @@ def forget(self, request):
     @lru_cache(maxsize=1)
     def get_policy(request: Request):
         """Pick the right policy based the request's path."""
-        # pylint:disable=too-many-return-statements
+        # pylint:disable=too-many-return-statements,too-complex
 
         path = request.path
 
         if path.startswith("/admin") or path.startswith("/googleauth"):
+            # Routes that require the Google auth policy
             return LMSGoogleSecurityPolicy()
 
+        if path.startswith("/dashboard/assignment/") or path.startswith(
```

Comment:
This method is becoming too complicated.

Looking into refactoring it to be more declarative:

https://github.com/hypothesis/lms/pull/6211